### PR TITLE
Add Optuna-Integration

### PR DESCRIPTION
**Pull request by gen740**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@conda-forge-admin, please ping conda-forge/help-python.

This recipe PR is ready for review.

File: recipes/optuna-integration/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 298219b3d8946f356b023ab95d704803bd90e1dd42264aa4ccd4a1acd56d1bf1
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/optuna-integration/meta.yaml:13]:
```suggestion
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/optuna-integration/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  number: 0
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (ocefpaf) [inline recipes/optuna-integration/meta.yaml:21]:
```suggestion
    - python >=3.6
    - setuptools >=61.1.0
    - pip
  run:
    - python >=3.6
```

**Comment by ocefpaf** [comment]
> * GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.

You checked this one but none of the maintainers listed commented here confirming.

**Comment by gen740** [comment]
@c-bata @contramundum53 @g-votte @HideakiImamura @knshnb @not522 @toshihikoyanase agree to maintain this with me?

**Comment by ocefpaf** [comment]
> @c-bata @contramundum53 @g-votte @HideakiImamura @knshnb @not522 @toshihikoyanase agree to maintain this with me?

Our instructions require a comment to capture that b/c +1 can be easily altered without an easy way to tracking the changes. I'll open an exception here but please do comment if you are submitting more recipes in the future.

PS: The main reason that we require this acknowledgment by all maintainers is b/c folks complained in the past about the automated emails they'll get from GH when added to the feedstock.

**Comment by ocefpaf** [comment]
We are still missing g-votte and ideakiImamura. @gen740 you can remove them here and re-add them in the feedstock if you wish. We cannot merge without their OK but you can when you have the feedstock commit rights.

**Comment by gen740** [comment]
Thank you for your guidance! I will proceed to remove g-votte and ideakiImamura from here.

**Pull request merged**